Phonebook.Frontend/src/app/modules/table/PersonsDataSource.ts

```diff
@@ -50,38 +53,41 @@ export class PersonsDataSource extends MatTableDataSource<Person> {
   public refresh(
     filterKeyword: string,
     searchFilters: SearchFilter[],
-    searchableColumns: Column[],
+    searchableColumns: ColumnId[],
     sort: TableSort
   ): Observable<Person[]> {
     this.loadingSubject.next(true);
     return new Observable<Person[]>(observer => {
-      let preResult = this.dataSource;
+      const searchParams: SearchParams = {
+        filterKeyword: filterKeyword,
+        searchFilters: searchFilters,
+        searchableColumns: searchableColumns,
+        sort: sort,
+        data: this.dataSource
+      };
 
-      // Filtering
-      searchFilters.forEach(searchFilter => {
-        preResult = TableLogic.filter(preResult, searchFilter.filterValue, [searchFilter.filterColumn]);
-      });
+      if (typeof Worker !== 'undefined' && runtimeEnvironment.environment != Environment.development) {
```

Comment:
why this must be disabled in development mode?

Reply:
Because Web Workers are not compiled with the Angular App in Development Mode so we have to provide the normal search. For Developing the WevWorker you have to compile the whole app... Kind of annoying.

Reply:
Switched it for some more general logic